plugins/module_utils/foreman_helper.py

```diff
@@ -1633,3 +1650,39 @@ def ensure_puppetclasses(module, entity_type, entity, expected_puppetclasses=Non
     'user_data',
     'ZTP',
 ]
+
+# interface specs
+interfaces_spec = dict(
+    id=dict(invisible=True),
+    mac=dict(),
+    ip=dict(),
+    ip6=dict(),
+    type=dict(choices=['interface', 'bmc', 'bond', 'bridge']),
+    name=dict(),
+    subnet=dict(type='entity'),
+    subnet6=dict(type='entity'),
+    domain=dict(type='entity', resource_type='subnets'),
```

Comment:
wrong `resource_type`. Have to be `domains` instead I would guess.

Reply:
good catch!